src/csharp/Microsoft.Spark.UnitTest/UdfSerDeTests.cs

```diff
@@ -5,20 +5,24 @@
 using System;
 using System.IO;
 using System.Reflection;
-using System.Runtime.Serialization.Formatters.Binary;
 using Microsoft.Spark.Utils;
 using Xunit;
+using MessagePack;
 
 namespace Microsoft.Spark.UnitTest
 {
     [Collection("Spark Unit Tests")]
     public class UdfSerDeTests
     {
-        [Serializable]
         private class TestClass
         {
             private readonly string _str;
 
+            // TODO: find out why MessagePack is requiring a parameterless constructor.
+            public TestClass()
+            {
+            }
```

Comment:
There are several more classes that require similar empty ctors, such as ExternalClass or Broadcast

Reply:
Now that [the standalone nuget package for BF](https://www.nuget.org/packages/System.Runtime.Serialization.Formatters/9.0.0-preview.6.24327.7) is now available, I'm leaning towards scraping the whole PR and moving to that instead, because this is IMO a major limitation that could break any code that uses user/3rd party classes that reference a class without an empty ctor. What do you think?

Reply:
Since microsoft doesn't add [Serializable] attributes to [new types since .NET 6](https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md),  sticking to NuGet will still not work long-term - As soon as there's a new commonly used type -  integration will fall apart.

Imho the best option long-term would be either to investigate MsgPack further (Custom resolver for classes without default ctor and workaround for the bug with empty stream), or, if there's no way to work-around ctor behavior - make serializer configurable and include both options

I created a PR with version bump up for .NET and dependencies, for .NET8 it's still possible to use BinaryFormatter from library, for .NET9 we'll have to migrate to NuGet